xbmc/pvr/addons/PVRClient.cpp

```diff
@@ -2211,6 +2209,31 @@ void CPVRClient::HandleAddonCallback(const char* strFunctionName,
   function(client);
 }
 
+template<typename F>
+void CPVRClient::HandleAddonCallback(const char* strFunctionName,
+                                     const void* kodiInstance,
+                                     F function,
+                                     bool bForceCall /* = false */)
+{
+  // Check preconditions.
+  auto* client{static_cast<const CPVRClient*>(kodiInstance)};
+  if (!client)
+  {
+    CLog::Log(LOGERROR, "{}: No instance pointer given!", strFunctionName);
+    return;
+  }
+
+  if (!bForceCall && client->m_bBlockAddonCalls && client->m_iAddonCalls == 0)
+  {
+    CLog::Log(LOGWARNING, LOGPVR, "{}: Ignoring callback from PVR client '{}'", strFunctionName,
+              client->ID());
+    return;
+  }
+
+  // Call.
+  function(client);
+}
```

Comment:
Just a suggestion, instead of creating an almost copy  of the function you could also change the existing function in this way:
```cpp
template<typename F, typename KodiInstance>
void CPVRClient::HandleAddonCallback(const char* strFunctionName,
                                     KodiInstance* kodiInstance,
                                     F function,
                                     bool bForceCall /* = false */)
{
  using Client = std::conditional_t<std::is_const_v<KodiInstance>, const CPVRClient, CPVRClient>;

  // Check preconditions.
  auto* client{static_cast<Client*>(kodiInstance)};
  if (!client)
  // ...
```

Reply:
Wow, that's cool. Did not know that. Thanks for pointing this out.